playjson/src/main/scala/io/finch/playjson/package.scala

```diff
@@ -11,10 +13,11 @@ package object playjson {
    * @tparam A the type of the data to decode into
    */
   implicit def decodePlayJson[A](implicit reads: Reads[A]): Decode.Json[A] =
-    // TODO: Eliminate toString conversion
-    // See https://github.com/finagle/finch/issues/511
-    // PlayJson can parse from byte[] automatically detecting the charset.
-    Decode.json((b, cs) => Try(Json.parse(BufText.extract(b, cs)).as[A]))
+    Decode.json({ (b, cs) =>
+      val buf = ChannelBufferBuf.Owned.extract(b)
+      if (buf.hasArray) Try(Json.parse(new String(buf.array(), 0, buf.readableBytes(), cs)).as[A])
```

Comment:
`Json.Parse` can take a `byte[]`, but when I convert that line to `Try(Json.parse(buf.array()).as[A])` it gives me the following in the 
```
Throw(spray.json.JsonParser$ParsingException: Unexpected character '\u0000'
```
I think I'm missing something about how the ChannelBufferBuf holds onto the underlying array.